src/web3/types.ts

```diff
@@ -0,0 +1,46 @@
+import { TypedDataDomain, TypedDataField } from 'ethers'
+import { CeloTx, EncodedTransaction } from '@celo/connect'
+
+// Generic wrapper interface for implementation-agnostic, lockable wallets used in Valora.
+// A "wallet" is responsible for managing an arbitrary number of "accounts" (i.e.,
+// unique addresses or signers) on a single blockchain. The methods defined here span
+// all the functionality currently expected of a "wallet" by the application.
+//
+// The types used for these methods are largely taken from Ethers where it was straightforward
+// to modify callsites to adhere to the new types. signTransaction is a notable deviation from
+// this due to its complexity.
+// TODO: Update signTransaction to use Ethers types.
+export interface ValoraWallet {
+  // Sign EIP712 typed data
+  signTypedData(
+    address: string,
+    domain: TypedDataDomain,
+    types: Record<string, Array<TypedDataField>>,
+    value: Record<string, any>,
+    primaryType: string
+  ): Promise<string>
+  signTransaction(tx: CeloTx): Promise<EncodedTransaction>
```

Comment:
Do you plan to make this method use more agnostic types?

Reply:
Yeah, that's the goal, though I'm punting on it for now since making the transaction object library-agnostic seems like a relatively difficult task.